src/actions/arcActions.ts

```diff
@@ -135,15 +135,38 @@ export function redeemProposal(daoAvatarAddress: string, proposalId: string, acc
   };
 }
 
-export function redeemReputationFromToken(scheme: Scheme, addressToRedeem: string) {
+export function redeemReputationFromToken(scheme: Scheme, addressToRedeem: string, redeemerAddress: Address, privateKey: string|undefined) {
   return async (dispatch: Redux.Dispatch<any, any>) => {
-    const observer = operationNotifierObserver(dispatch, "Redeem reputation");
-    const reputationFromTokenScheme = scheme.ReputationFromToken as ReputationFromTokenScheme;
+    const arc = getArc();
+    if (privateKey) {
+      console.log("SIGNING WITH PK");
+      const reputationFromTokenScheme = scheme.ReputationFromToken as ReputationFromTokenScheme;
```

Comment:
shouldnt this all be in the client library?

Reply:
well, yes, you are probably right - it started as a one-liner but then I needed more. Also it needs to be deployed quickly, so I chose the quick solution.